src/zkevm_specs/evm_circuit/execution/create.py

```diff
@@ -0,0 +1,186 @@
+    )
+
+    # Verify depth is less than 1024
+    instruction.range_lookup(depth, CALL_CREATE_DEPTH)
+
+    # ErrNonceUintOverflow constraint
+    (is_not_overflow, _) = instruction.compare(nonce, nonce_prev, N_BYTES_U64)
+    instruction.is_zero(is_not_overflow)
+
+    # add contract address to access list
+    instruction.add_account_to_access_list(tx_id, contract_address)
+
+    # ErrContractAddressCollision constraint
+    # code_hash_prev could be either 0 or EMPTY_CODE_HASH
+    # code_hash should be EMPTY_CODE_HASH to make sure the account is created properly
+    code_hash, code_hash_prev = instruction.account_write(
+        contract_address, AccountFieldTag.CodeHash
+    )
+    instruction.constrain_in(code_hash_prev, [FQ(0), RLC(EMPTY_CODE_HASH).expr()])
+    instruction.constrain_equal(code_hash, RLC(EMPTY_CODE_HASH))
+
+    # Propagate is_persistent
+    callee_reversion_info = instruction.reversion_info(call_id=callee_call_id)
+    instruction.constrain_equal(
+        callee_reversion_info.is_persistent,
+        reversion_info.is_persistent * is_success,
+    )
+
+    # can't be a STATICCALL
+    instruction.is_zero(is_static)
+
+    # transfer value from caller to contract address
+    instruction.transfer(caller_address, contract_address, value, callee_reversion_info)
+
+    # gas cost of memory expansion
+    (next_memory_size, memory_expansion_gas_cost,) = instruction.memory_expansion(
+        offset,
+        size,
+    )
+
+    # CREATE = GAS_COST_CREATE + memory expansion + GAS_COST_CODE_DEPOSIT * len(byte_code)
+    # CREATE2 = gas cost of CREATE + GAS_COST_COPY_SHA3 * memory_size
+    # byte_code is only available in `return_revert`,
+    # so the last part (GAS_COST_CODE_DEPOSIT * len(byte_code)) won't be calculated here
+    gas_left = instruction.curr.gas_left
+    gas_cost = GAS_COST_CREATE + memory_expansion_gas_cost
+    if is_create2 == 1:
+        memory_size, _ = instruction.constant_divmod(size.expr() + 31, FQ(32), N_BYTES_MEMORY_SIZE)
+        gas_cost += GAS_COST_COPY_SHA3 * memory_size
+    gas_available = gas_left - gas_cost
+
+    # Apply EIP 150
+    one_64th_gas, _ = instruction.constant_divmod(gas_available, FQ(64), N_BYTES_GAS)
+    all_but_one_64th_gas = gas_available - one_64th_gas
+    is_u64_gas = instruction.is_zero(
+        instruction.sum(instruction.rlc_encode(gas_left, N_BYTES_GAS).le_bytes[N_BYTES_GAS:])
+    )
+    callee_gas_left = instruction.select(
+        is_u64_gas,
+        instruction.min(all_but_one_64th_gas, gas_left, N_BYTES_GAS),
+        all_but_one_64th_gas,
+    )
+
+    # copy init_code from memory to bytecode
+    copy_rwc_inc, _ = instruction.copy_lookup(
+        instruction.curr.call_id,  # src_id
+        CopyDataTypeTag.Memory,  # src_type
+        instruction.next.code_hash.expr(),  # dst_id
+        CopyDataTypeTag.Bytecode,  # dst_type
+        offset,  # src_addr
+        offset + size,  # src_addr_boundary
+        FQ(1),  # dst_addr
```

Comment:
I couldn't figure out why is this starting with `1`, is there any specific reason for this?

Reply:
my mistake, it's always failed when I set `0`.Just tried to make it work first, and I was planning to ask you after your Vietnam trip but I totally forgot this. 

I saw your comment [below](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/files#r1166300624),  that might be the reason why. I'll fix it and request you to review this again. thanks!

Reply:
fixed in [8c49721](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/8c4972148e3ca79d57e50711498be2170590fc59)